tests/unit/connection_parameters_tests.py

```diff
@@ -695,6 +695,34 @@ def test_uses_default_virtual_host_if_only_slash_is_specified(self):
         self.assertEqual(parameters.virtual_host,
                          pika.URLParameters.DEFAULT_VIRTUAL_HOST)
 
+    def test_uses_default_virtual_host_via_encoded_slash_1(self):
+        parameters = pika.URLParameters('amqp://myserver.mycompany.com/%2F')
+        self.assertEqual(parameters.virtual_host,
+                         pika.URLParameters.DEFAULT_VIRTUAL_HOST)
+
+    def test_uses_default_virtual_host_via_encoded_slash_2(self):
```

Comment:
I think it may be better to use more descriptive test names instead of suffixing with 1,2,3,4. It seems like the variability is upper case vs lower case encoding and terminating slash versus no terminating slash.